velox/functions/prestosql/HyperLogLogFunctions.h

```diff
@@ -81,4 +82,57 @@ struct EmptyApproxSetWithMaxErrorFunction {
     return true;
   }
 };
+
+template <typename T>
+struct MergeHllFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  std::shared_ptr<memory::MemoryPool> pool_;
+  std::unique_ptr<HashStringAllocator> allocator_;
+  std::unique_ptr<aggregate::prestosql::HllAccumulator<int64_t, true>>
+      accumulator_;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& /*config*/,
+      const arg_type<Array<HyperLogLog>>* /*input*/) {
+    pool_ = memory::memoryManager()->addLeafPool();
+    allocator_ = std::make_unique<HashStringAllocator>(pool_.get());
+    accumulator_ =
+        std::make_unique<aggregate::prestosql::HllAccumulator<int64_t, true>>(
+            allocator_.get());
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<HyperLogLog>& result,
+      const arg_type<Array<HyperLogLog>>& input) {
+    // Reset and re-create allocator for reuse
+    allocator_->clear();
```

Comment:
This is still pretty heavy-weight. HSA is not designed to be used like this.

I suggest we modify HllAccumulator to add a template parameter for memory allocator and use standard memory allocator here and HSA-based one in the aggregation.

CC: @xiaoxmeng